clinvar/src/main/scala/org/broadinstitute/monster/etl/clinvar/models/output/SCVTrait.scala

```diff
@@ -110,24 +110,26 @@ object SCVTrait {
         val xrefMatch = {
           val isXrefMapping = candidateMapping.mappingType == "XRef"
           val xrefMatches = xrefs.exists { xref =>
-            xref.obj(Str("db")).str == candidateMapping.mappingRef &&
-            xref.obj(Str("id")).str == candidateMapping.mappingValue
+            xref.db == candidateMapping.mappingRef &&
+            xref.id == candidateMapping.mappingValue
           }
 
           isXrefMapping && xrefMatches
         }
 
         sameTraitType && (nameMatch || xrefMatch)
       }
+
       // Find the MedGen ID / name to look for in the VCV traits.
       val matchingMedgenId = matchingMapping.flatMap(_.medgenId)
       val matchingName = matchingMapping.flatMap(_.medgenName)
       // Find the VCV trait with the matching MedGen ID if it's defined.
       // Otherwise match on preferred name.
-      val matchingVcvTrait =
-        matchingMedgenId.fold(traits.find(_.name == matchingName)) { id =>
-          traits.find(_.medgenId.contains(id))
-        }
+      val matchingVcvTrait = medgenDirectMatches.headOption
+        .orElse(xrefDirectMatches.headOption)
+        .orElse(traits.find(_.medgenId == matchingMedgenId))
+        .orElse(traits.find(_.name == matchingName))
```

Comment:
The way this is structured, we're still searching through all the trait mappings even if we find a direct xref match. Could you pull the mapping-related logic into a separate method, so we can call it like:
```scala
val matchingVcvTrait = medgenDirectMatch
  .orElse(xrefDirectMatch)
  .orElse(findTraitMappingMatch(... args ...))
```